R/zchunk_LA144.building_det_flsp.R

```diff
@@ -46,60 +48,379 @@ module_energy_LA144.building_det_flsp_DISABLED <- function(command, ...) {
+         # Divide floorspace by population to get per capita floorspace, and extrapolate to all historical years
+    CEDB_ResFloorspace_chn %>%
+      gather(year, value_flsp, -country, -iso) %>% # Convert to long form
+      mutate(year = as.integer(year)) %>% # Needs to be integer to combine with L100.Pop_thous_ctry_Yh
+      filter(year %in% HISTORICAL_YEARS) %>% # Ensure within historical time period
+      left_join_error_no_match(L100.Pop_thous_ctry_Yh, by = c("iso", "year")) %>%
+      # Divide floorspace by population to get per capita floorspace
+      mutate(value_pcflsp = value_flsp / value * CONV_BIL_THOUS) %>% # Note: converting to thousand m2 because population is in thousands
+      select(iso, year, value_pcflsp) %>%
+      group_by(iso) %>%
+      # Expand table to include all historical years
+      complete(year = HISTORICAL_YEARS) %>%
+      # Extrapolate to fill out values for all years
+      # Rule 2 is used so years outside of min-max range are assigned values from closest data, as opposed to NAs
+      mutate(value_pcflsp = approx_fun(year, value_pcflsp, rule = 2)) %>%
+      ungroup() ->
+      L144.pcflsp_m2_chn_Yh
+
+    # Odyssee_ResFloorspacePerHouse provides residential floorspace (m2) per house (not person) from 1980 to 2009 for 29 countries
+         # A44.HouseholdSize provides number of persons/dwelling data, which will be used to calculate floorspace per capita
+
+    # We need to prepare some lists and reshape tables first
+    # First, convert household data to long form so it can be joined at a later step
+    A44.HouseholdSize_long <- A44.HouseholdSize %>%
+      select(-Variable, -Unit) %>%
+      gather(year, value_pcdwelling)
+
+    # IEA_PCResFloorspace provides residential floorspace (m2) per person for 16 selected countries for 1980 to 2004
+         # Note that IEA data will be chosen over Odyssee for duplicate countries b/c it reports per capita instead of per house
+
+    # Reshape IEA data to long form
+    IEA_PCResFloorspace_long <- gather(IEA_PCResFloorspace, year, value_pcflsp, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year),
+             value_pcflsp = as.numeric(value_pcflsp))
+
+    # Create list of IEA iso's. It will be used to remove these iso's from Odyssee data.
+    # Also, create a list of years from IEA data in order to be used for filtering.
+    list_iso_IEA <- unique(IEA_PCResFloorspace_long$iso)
+    list_years_IEA <- unique(IEA_PCResFloorspace_long$year)
+
+    Odyssee_ResFloorspacePerHouse %>%
+      gather(year, value_phflsp, matches(YEAR_PATTERN)) %>% # Convert to long form
+      filter(!iso %in% list_iso_IEA) %>% # Remove iso's that are in IEA dataset
+      # left_join_error_no_match cannot be used because joining table does not contain every year, which will introduce NAs
+      left_join(A44.HouseholdSize_long, by = "year") %>%
+      mutate(year = as.integer(year)) %>% # Convert years to integer (or numeric) to be able to extrapolate
```

Comment:
I had it there because `A44.HouseholdSize_long` also had year as characters. So it was to kill two birds with one stone. In case it made it clearer though, I did as you suggested and converted year for `A44.HouseholdSize_long` above.